echopop/computation/biology.py

```diff
@@ -104,160 +128,751 @@ def sum_strata_weight( haul_data: pd.DataFrame ,
+                                                                                   on = [ 'stratum_num' , 'sex' ] , 
+                                                                                   how = 'left' )
     
-    ### Normalize the age-length-sex indexed proportions
-    dist_weight_sum = ( 
-       weight_length_age_sex_stratum 
-        .merge( weight_strata , on = [ 'stratum_num' ] ) 
-        .groupby( [ 'stratum_num' , 'sex' ] ) 
-        .apply( lambda df: pd.Series( {  
-            'proportion_normalized': ( df.proportion_weight_all * ( df.summed_weight_all / df.weight_stratum_total ).sum( ) ).sum( ) 
-        } ) ) 
-        .reset_index( ) 
-    ) 
+    # ---- Normalized weight proportions for all fish
+    distributed_aged_weight_proportions[ 'normalized_proportion_weight_all' ] = (
+        distributed_aged_weight_proportions.proportion_weight_sex_all * distributed_aged_weight_proportions.proportion_weight_all
+    )
 
-    ### Carriage return
-    return dist_weight_sum
+    # ---- Normalized weight proportions for jus adult fish
+    distributed_aged_weight_proportions[ 'normalized_proportion_weight_adult' ] = (
+        distributed_aged_weight_proportions.proportion_weight_sex_adult * distributed_aged_weight_proportions.proportion_weight_adult
+    )   
+
+    # ---- Remove unnecessary columns and return the dataframe
+    return distributed_aged_weight_proportions.filter( regex = '^(?!weight_|total_weight_|proportion_).*' )
 
-def calculate_aged_proportions( weight_length_age_sex_stratum: pd.DataFrame ,
-                                weight_strata: pd.DataFrame ):
+def calculate_aged_biomass( kriging_biomass_df: pd.DataFrame ,
+                            specimen_data: pd.DataFrame ,
+                            length_distribution: pd.DataFrame ,
+                            age_distribution: pd.DataFrame ,
+                            aged_proportions: pd.DataFrame ):
+    """
+    Calculate the kriged biomass distributed over length and age for each sex and among
+    all fish (sexed)
+
+    Parameters
+    ----------
+    kriging_biomass_df: pd.DataFrame
+        Dataframe containing kriged biomass estimates
+    specimen_data: pd.DataFrame
+        Dataframe containing specimen weights
+    length_distribution: np.ndarray
+        Array containing length bins/intervals
+    age_distribution: np.ndarray
+        Array containing age bins/intervals
+    aged_proportions: pd.DataFrame
+        Dataframe containing the weight proportions of aged fish within each stratum
+    """
+
+    ### Sum aged fish weights across age and length bins, then calculate weight proportions
+    proportions_weight_length_age_sex = compute_aged_weight_proportions( specimen_data ,
+                                                                         length_distribution ,
+                                                                         age_distribution )
```

Comment:
Add to the comment to make it specific that it is for age and length weight proportion _within_ each sex. Otherwise the output variable name does not make sense.

You might want to consider renaming `proportions_weight_length_age_sex` to something like `proportions_weight_length_age_within_each_sex` so that there is no chance of confusion.
```suggestion
    ### Sum aged fish weights across age and length bins for each sex, then calculate weight proportions within each sex
    proportions_weight_length_age_sex = compute_aged_weight_proportions( specimen_data ,
                                                                         length_distribution ,
                                                                         age_distribution )
```